src/app/load-app.js

```diff
@@ -1,16 +1,37 @@
+import { useDataQuery } from '@dhis2/app-runtime'
 import i18n from '@dhis2/d2-i18n'
 import { CircularLoader, CenteredContent, NoticeBox } from '@dhis2/ui'
 import { node } from 'prop-types'
 import React from 'react'
 import { useCustomFormsPrefetch } from '../custom-forms/index.js'
-import { useMetadata } from '../shared/index.js'
+import { useMetadata, useUserInfoStore } from '../shared/index.js'
 import css from './load-app.module.css'
 
+const query = {
+    user: {
+        resource: 'me',
+        params: {
+            fields: ['authorities', 'avatar', 'email', 'name', 'settings'],
+        },
+    },
+}
+
 const LoadApp = ({ children }) => {
     useCustomFormsPrefetch()
+    const updateStore = useUserInfoStore((state) => state.setAuthorities)
+    const {
+        loading: userLoading,
+        error: userError,
+        data: userData,
+    } = useDataQuery(query, {
+        onComplete: (userData) => {
+            updateStore(userData.user?.authorities)
+        },
+    })
+
```

Comment:
I think we should try to use `tanstack/query` here, like everywhere else in the app.
It uses `app-runtime` `engine` under the hood for the actual request, and I think we should manage to make them the same? Or does that apply headers or something we cannot reuse the cached-response?

Reply:
Sounds good. I thought maybe we wanted to only use tanstack query for the requests we wanted to more actively handle caching strategy for, but if we're okay with using it for this, that sounds good, and it will eliminate the need for the zustand store.

Commit here to refactor with tanstack query: https://github.com/dhis2/data-entry-app/commit/1dc468048105ef3136aa04b33278f83c4c8fe8b3